### PR TITLE
Interactivity exception consistency

### PR DESCRIPTION
**Pull request by VelvetThePanda**
# Summary
Cleans up checks in interactivity and makes all interactivity methods as consistent as possible (the exception being methods that take an Id, or collection of components, which have additional checks).

# Details
This should fix some regressions in interactivity involving faulty checks that would cause exceptions to be thrown on valid messages. 

# Changes proposed
- Make interactivity component checks consistent

# Notes
@IDoEverything do **not** merge this PR until at least 3 people have approved changes. I'm not responsible if it breaks.

### CODE REVIEW

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 1]
```diff
@@ -257,8 +267,11 @@ public async Task<InteractivityResult<ComponentInteractionCreateEventArgs>> Wait
             if (message.Author != this.Client.CurrentUser)
                 throw new InvalidOperationException("Interaction events are only sent to the application that created them.");
 
-            if (message.Components is null || !message.Components.Any(a => a.Components.Any(c => c is DiscordButtonComponent)))
-                throw new ArgumentException("Message does not contain any buttons.");
+            if (!message.Components.Any())
+                throw new ArgumentException("Message doesn't contain any components.");
+
+            if (!message.Components.SelectMany(c => c.Components).Any(c => c.Type is ComponentType.Button))
+                throw new ArgumentException("Message does not contain any select components.");
```
Comment (CloudTheWolf) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:274]: Would expect this to to say  "Message does not contain any button components." to be consistent.

Comment (VelvetThePanda) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:274]: Oops, forgot to change that

**Review by CloudTheWolf — APPROVED** [review]
All feedback addressed, no other issues found. (1/3 requested approvals)

**Comment by IDoEverything** [comment]
@VelvetThePanda there's merge conflicts

**Comment by VelvetThePanda** [comment]
@IDoEverything fixed

**Review by IDoEverything — APPROVED** [review]

**Pull request merged**